app/models/missing_link_finder.rb

```diff
@@ -10,8 +10,30 @@ def initialize(redirection)
   end
 
   def run
-    html = fetch(redirection.url)
+    begin
+      html = fetch(redirection.url)
+      if html.nil?
```

Comment:
This means the status wasn't successful (4xx or 5xx response code).

Reply:
Would it make sense to actually make the status part of the conditional? Or maybe we could move to a response type or something? `html.nil?` feels very implicit, but maybe I'm over thinking it.

Reply:
>Would it make sense to actually make the status part of the conditional?

Oh yeah, great idea. `fetch` is hiding too much here. I'll open it up.